microsetta_interface/implementation.py

```diff
@@ -1464,14 +1639,13 @@ def get_source(*, account_id=None, source_id=None):
     if has_error:
         return account_output
 
-    has_error, consent_required, _ = ApiRequest.get(
-        '/accounts/%s/source/%s/consent/%s' % (account_id, source_id, 'data'))
+    need_reconsent = check_current_consent(account_id, source_id, "data")
 
-    if has_error:
-        return consent_required
-
-    if consent_required["result"]:
-        return render_consent_page(account_id, source_id, "data")
+    if not session.get(ADMIN_MODE_KEY, False) and need_reconsent\
+            and not session.get(RECONSENT_DECLINED_KEY, False):
```

Comment:
```suggestion
    if session.get(ADMIN_MODE_KEY, False) is False:
        if need_reconsent:
            if session.get(RECONSENT_DECLINED_KEY, False) is False:
```
similar to above but i won't die on this hill

Reply:
Adjusted.